runtime/core.go

```diff
@@ -253,6 +253,29 @@ func Hash(f *Frame, o *Object) (*Int, *BaseException) {
 	return toIntUnsafe(h), nil
 }
 
+// Hex returns the result of o.__hex__ if defined.
+func Hex(f *Frame, o *Object) (*Object, *BaseException) {
+	hex := o.typ.slots.Hex
+	if hex == nil {
+		if method, raised := o.typ.mroLookup(f, NewStr("__hex__")); raised != nil {
```

Comment:
No need to lookup `__hex__`. If that method exists, an automatically generated Hex slot will exist and forward to `__hex__`.